core/architecture.md

```diff
@@ -0,0 +1,82 @@
+# Nimbess Controller Architecture
+
+## Overview
+
+Nimbess is a light-weight distributed controller architecture that runs as a
+Kubernetes pod across every Kubelet node. Nimbess by default supports the
+Berkley Extensible Software Switch [BESS](https://github.com/NetSys/bess)
+dataplane in order to achieve maximum network pipeline programmability and
+performance. The primary goal of Nimbess is to change the approach to the
+networking paradigm in Kubernetes itself by implementing Unified Network Policy
+across Kubernetes. More on how this works can be found in the 
+[Functional Design Guide](functional-spec.md). The purpose of this document is
+to describe the software architecture and design.
+
+## Background
+
+Typical CNI or networking plugin designs for Kubernetes either rely on an SDN
+controller/dataplane combination and/or multiple external applications such
+as iptables, haproxy/envoy, conntrack, etc in order to achieve a fully featured
+network dataplane within Kubernetes. These features include services, ingress,
+and also "network policy". The Network Policy resource in Kubernetes is limited
+and refers only to access-control policy of "who can talk to who". 
+
+One of the main goals of Nimbess is to simplify and consolidate these features
+into a single point of entry using a custom resource definition (CRD) called
```

Comment:
I think our stated main goal should not be to consolidate features, but to provide an entirely new way of thinking about container or kubernetes networking that's based on policy and intent and defining a mechanism to express and realize them.

Reply:
yes, will reword it